Dijkstra.java

```diff
@@ -0,0 +1,208 @@
+            for (int j = 0; j < codes.size(); j++){
+                temp.add(distancesIn.nextInt());
+            }
+            distances.add(temp);
+        }
+
+        // populate the unvisited queue
+        for (int i = 0; i < codes.size(); i ++){
+            Node temp = new Node (codes.get(i));
+            unvisited.add(temp);
+        }
+
+        // always close your scanners
+        codesIn.close();
+        distancesIn.close();
+
+        // get the airport codes from the user
+        System.out.print("\nStart: ");
+        start = input.nextLine();
+        System.out.print("\nEnd: ");
+        end = input.nextLine();
+
+        // visit the first node
+        currentNode = unvisited.get(codes.indexOf(start));
+        currentNode.setDistance(0);
+        visit(currentNode);
+        unvisited.remove(currentNode);
+        visited.add(currentNode);
+
+        // visit the rest of the nodes
+        while(!endFlag) {
+            currentNode = getNext(unvisited);
+            visit(currentNode);
+            unvisited.remove(currentNode);
+            visited.add(currentNode);
+            if (currentNode.getCode().equals(end)){
+                endFlag = true;
+            }
+        }
+
+
+        // output results to the user on the console
+        System.out.println("The shortest path between " + start + " and " + end + " is " + currentNode.getDistance() + " miles.");
+        System.out.println("The shortest path is:");
+        ArrayList path = getPath(visited, start, end);
+        for (int i = path.size() - 1; i > 0; i--){
+            System.out.print(path.get(i) + " --> ");
+        }
+        System.out.print(end);
+
+        // close the keyboard scanner
+        input.close();
+    }
+
+    // visit a Node in the graph. Process the corresponding row in the adjacency matrix
+    // and adjust distances as necessary.
+    // if the distance is adjusted, set the predecessor node to the node being processed
+    public static void visit (Node current){
+        int rowNum = codes.indexOf(current.getCode());
+        ArrayList <Integer> row = distances.get(rowNum);
+        for(Node temp : unvisited){
+            if ((current.getDistance() + row.get(codes.indexOf(temp.getCode()))) < temp.getDistance()) {
+                temp.setDistance(current.getDistance() + row.get(codes.indexOf(temp.getCode())));
+                temp.setPrevNode(current);
+            }
+        }
+    }
+
+    // get the smallest element in the list (next node to visit)
+    public static Node getNext (ArrayList<Node> list){
+        Node temp = list.get(0);
+        for (Node elm : list){
+            if (elm.compareTo(temp) < 1){
+                temp = elm;
+            }
+        }
+        return temp;
+    }
+
+    // generate the path by tracing each node to its predecessor
+    public static ArrayList<String> getPath(ArrayList<Node> n, String st, String en){
+        Node e = null;
+        Node s = null;
```

Comment:
Naming variables one letter does make the code _look_ cleaner, but in reality, it severely reduces readability. I would go and name these variables something descriptive so that it is easy to tell what is going on.